src/models/meta_details.rs

```diff
@@ -189,6 +201,55 @@ fn library_item_sync(library_item: &Option<LibraryItem>, profile: &Profile) -> E
     }
 }
 
+fn selected_override_update(
+    selected: &mut Option<Selected>,
+    meta_items: &Vec<ResourceLoadable<MetaItem>>,
+) -> Effects {
+    let meta_path = if let Some(Selected {
+        meta_path,
+        stream_path: None,
+    }) = &selected
+    {
+        meta_path
+    } else {
+        return Effects::default();
+    };
```

Comment:
Nitpick: This is why in Rust we have a `match`, it doesn't make sense to use `if let` here.
```suggestion
    let meta_path = match &selected {
        Some(Selected { meta_path, .. }) => meta_path,
        None => return Effects::default(),
    }
```